sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_lease_async.py

```diff
@@ -3,76 +3,80 @@
-        lease has yet been acquired.
-    :ivar str etag:
-        The ETag of the lease currently being maintained. This will be `None` if no
-        lease has yet been acquired or modified.
-    :ivar ~datetime.datetime last_modified:
-        The last modified timestamp of the lease currently being maintained.
-        This will be `None` if no lease has yet been acquired or modified.
-
     :param client:
         The client of the file system, directory, or file to lease.
-    :type client: ~azure.storage.filedatalake.aio.FileSystemClient or
-        ~azure.storage.filedatalake.aio.DataLakeDirectoryClient or ~azure.storage.filedatalake.aio.DataLakeFileClient
+    :type client:
+        ~azure.storage.filedatalake.aio.FileSystemClient or
+        ~azure.storage.filedatalake.aio.DataLakeDirectoryClient or
+        ~azure.storage.filedatalake.aio.DataLakeFileClient
     :param str lease_id:
         A string representing the lease ID of an existing lease. This value does not
         need to be specified in order to acquire a new lease, or break one.
     """
-    def __init__(
-            self, client, lease_id=None
-    ):  # pylint: disable=missing-client-constructor-parameter-credential,missing-client-constructor-parameter-kwargs
-        # type: (Union[FileSystemClient, DataLakeDirectoryClient, DataLakeFileClient], Optional[str]) -> None
-        super(DataLakeLeaseClient, self).__init__(client, lease_id)
+
+    id: str
+    """The ID of the lease currently being maintained. This will be `None` if no
+        lease has yet been acquired."""
+    etag: Optional[str]
+    """The ETag of the lease currently being maintained. This will be `None` if no
+        lease has yet been acquired or modified."""
+    last_modified: Optional["datetime"]
+    """The last modified timestamp of the lease currently being maintained.
+        This will be `None` if no lease has yet been acquired or modified."""
+
+    def __init__(  # pylint: disable=missing-client-constructor-parameter-credential, missing-client-constructor-parameter-kwargs
+        self, client: Union["FileSystemClient", "DataLakeDirectoryClient", "DataLakeFileClient"],
+        lease_id: Optional[str] = None
+    ) -> None:
+        self.id = lease_id or str(uuid.uuid4())
+        self.last_modified = None
+        self.etag = None
 
         if hasattr(client, '_blob_client'):
-            _client = client._blob_client  # type: ignore
+            _client = client._blob_client
         elif hasattr(client, '_container_client'):
-            _client = client._container_client  # type: ignore
+            _client = client._container_client
         else:
             raise TypeError("Lease must use any of FileSystemClient DataLakeDirectoryClient, or DataLakeFileClient.")
 
         self._blob_lease_client = BlobLeaseClient(_client, lease_id=lease_id)
 
-    def __enter__(self):
+    def __enter__(self) -> None:
```

Comment:
We should delete `__enter__` and `__exit__` from the async version since they are sync functions. Yes, this is technically a breaking change but I think in this case its ok. I think we did this for file-share.

Reply:
Yup looking back at the large typing PR for `azure-storge-file-share`, there was no `__enter__` nor `__exit__` for lease client, so this seems reasonable to remove.